pallets/funding/src/types.rs

```diff
@@ -775,6 +780,18 @@ pub mod inner_types {
 		pub evaluators_outcome: EvaluatorsOutcome<Balance>,
 	}
 
+	// putting the enum inside a struct in case we need to extend it with other fields later on
+	#[derive(Clone, Encode, Decode, Eq, PartialEq, RuntimeDebug, TypeInfo, MaxEncodedLen)]
+	pub struct AuctionRoundInfo<Balance> {
+		// Will be `Some` after the auction closing phase ends and winning bids are calculated
+		pub is_oversubscribed: IsOversubscribed<Balance>,
+	}
+	#[derive(Clone, Encode, Decode, Eq, PartialEq, RuntimeDebug, TypeInfo, MaxEncodedLen)]
+	pub enum IsOversubscribed<Balance> {
+		Yes { total_usd_bid: Balance },
+		No,
+	}
```

Comment:
I think I would prefer an Option here instead of the IsOversubscribed enum. I think you had an option as well looking at the comments.

Reply:
You could rename the AuctionRoundInfo (as it only holds the is_oversubscribed inner value). So we sort of unpack the inner value as the outer value, e.g. AuctionRoundInfo -> IsOversubscribed (or some other name), which is an option. If we add a clear doc comment on the value, I think it would be clear enough what the Some(value) means.

Reply:
So instead of
`pub auction_round_info: AuctionRoundInfo,`
we do:
```
/// Whether or not the auction is oversubscribed. An auction is oversubscribed if the first bucket has been filled. 
/// In this case we have to calculate the weighted average price.
/// If the auction is oversubscribed we store the usd total of all accepted bids.
/// We use this value for the wap calculation. 
pub auction_oversubscribed: Option<Balance>,
```